src/Npgsql/TypeHandlers/CompositeHandlers/MappedCompositeHandler.cs

```diff
@@ -75,14 +90,16 @@ public override int ValidateAndGetLength(T value, ref NpgsqlLengthCache? lengthC
             lengthCache.Set(0);
 
             // number of fields + (type oid + field length) * member count
-            var length = sizeof(int) + sizeof(int) * 2 * _members.Length;
-            foreach (var member in _members)
+            var length = sizeof(int) + sizeof(int) * 2 * _memberHandlers.Length;
+            foreach (var member in _memberHandlers)
                 length += member.ValidateAndGetLength(value, ref lengthCache);
 
             return lengthCache.Lengths[position] = length;
         }
 
-        public static MappedCompositeHandler<T> Create(PostgresCompositeType pgType, ConnectorTypeMapper typeMapper, INpgsqlNameTranslator nameTranslator)
+        CompositeMemberHandler<T>[] CreateMemberHandlers() => CreateMemberHandlers((PostgresCompositeType)PostgresType, _typeMapper, _nameTranslator);
```

Comment:
nit: for extra cleanliness we could hide base.PostgresType with a typed new property:

```c#
new PostgresCompositeType PostgresType => (PostgresCompositeType)base.PostgresType;
```

Reply:
A nice idea. I will use it twice (am working on `ctor` support), so it makes sense.

Reply:
Will do in the next PR since there are some changes in handler. So probably this property would be removed.